mountpoint-s3/src/prefetch/parquet_prefetch.rs

```diff
@@ -0,0 +1,155 @@
+use crate::sync::Arc;
+use bytes::{Bytes, BytesMut};
+use futures::pin_mut;
+use futures::StreamExt;
+use intervaltree::IntervalTree;
+use mountpoint_s3_client::types::ETag;
+use mountpoint_s3_client::ObjectClient;
+use parquet::errors::ParquetError;
+use parquet::file::footer::decode_footer;
+use parquet::file::metadata::ParquetMetaData;
+use tracing::trace;
+
+use super::PrefetchReadError;
+
+const PARQUET_MAGIC_LEN: usize = 8;
+
+pub async fn read_parquet_metadata<Client>(
+    client: Arc<Client>,
+    bucket: &str,
+    key: &str,
+    if_match: &ETag,
+    total_size: u64,
+) -> Result<(Bytes, std::ops::Range<u64>), PrefetchReadError<Client::ClientError>>
+where
+    Client: ObjectClient + Send + Sync + 'static,
```

Comment:
I'd rename `total_size` to `object_size`.

```suggestion
pub async fn read_parquet_metadata<Client>(
    client: Arc<Client>,
    bucket: &str,
    key: &str,
    if_match: &ETag,
    object_size: u64,
) -> Result<(Bytes, std::ops::Range<u64>), PrefetchReadError<Client::ClientError>>
where
    Client: ObjectClient + Send + Sync + 'static,
```

And again for `fetch_metadata`.